Remove dependency on `golang.org/x/sys`

### Operating System

- [X] Windows
- [ ] macOS
- [ ] Linux
- [ ] FreeBSD
- [ ] Android
- [ ] iOS

### What feature would you like to be added?

Remove dependency on `golang.org/x/sys`. Aim to be zero dependencies.

### Why is this needed?

> The discussion was started from Discord chat: https://discord.com/channels/842049801528016967/1123106378731487345/1275998637826248744

For now, the only dependency is `golang.org/x/sys`.

If we can remove the dependency, we can embed purego into more situation like Golang toolchain.

Currently I'm working on a go fork and try to introduce our own C lib into Go toolchain, if purego can have zero dependencies, we can import the C lib without CGO (instead, use purego for sure). But in the toolchain, dependency chain has limitation that `golang.org/x/sys` is not allowed to be here. :(

> What kind of commands did you execute? I'd like to know the details.

Here's the steps I took:

1. Add purego to my module A with `go get github.com/ebitengine/purego@5905af50730466425f56454a0491ae991d644beb`
2. Add some usage into module A, to simpify the process, I copied the example code into the init function, like this:
```
package module_a

import (
	"github.com/ebitengine/purego"
)

func init() {
	libc, err := purego.Dlopen("/usr/lib/libSystem.B.dylib", purego.RTLD_NOW|purego.RTLD_GLOBAL)
	if err != nil {
		panic(err)
	}
	var puts func(string)
	purego.RegisterLibFunc(&puts, libc, "puts")
	puts("Calling C from Go with purego!")
}
```
3. Add module A into golang toolchain 1.22.5 with `go get private.pkgs/module_a@commitid`, and get the output:
```
go: downloading private.pkgs/module_a v0.1.0-alpha.3.0.20240828062439-2e1170466358
go: added github.com/ebitengine/purego v0.8.0-alpha.5
go: upgraded private.pkgs/module_a v0.1.0-alpha.3.0.20240815033454-aa6aed5d72d1 => v0.1.0-alpha.3.0.20240828062439-2e1170466358
go: upgraded golang.org/x/sys v0.15.0 => v0.23.0
```
4. Add some calls to module A into `go/src/crypto/internal`
5. Run `go mod vendor`, get a wired error:
```
go: std/crypto/tls imports
	golang.org/x/crypto/chacha20poly1305 imports
	golang.org/x/sys/cpu: missing go.sum entry for module providing package golang.org/x/sys/cpu (imported by golang.org/x/crypto/chacha20poly1305); to add:
	go get golang.org/x/crypto/chacha20poly1305@v0.16.1-0.20231129163542-152cdb1503eb
```
6. Run `go get golang.org/x/sys/cpu` to get missing dependencies:
```
go: downloading golang.org/x/sys v0.24.0
go: upgraded golang.org/x/sys v0.23.0 => v0.24.0
```
7. Build the toolchain with `bash make.bash`
8. Run the test from toolchain with `bash run.bash --no-rebuild`, and get the output with this (skipped some output):
```
...
--- FAIL: TestDependencies (1.46s)
    deps_test.go:765: unexpected dependency: github.com/ebitengine/purego imports [errors fmt github.com/ebitengine/purego/internal/cgo github.com/ebitengine/purego/internal/fakecgo github.com/ebitengine/purego/internal/strings math reflect runtime runtime/cgo sync syscall unsafe]
    deps_test.go:765: unexpected dependency: github.com/ebitengine/purego/internal/cgo imports [C errors unsafe]
    deps_test.go:765: unexpected dependency: github.com/ebitengine/purego/internal/fakecgo imports [syscall unsafe]
    deps_test.go:765: unexpected dependency: github.com/ebitengine/purego/internal/strings imports [unsafe]
    deps_test.go:765: unexpected dependency: gitlab.ikv.tw/apsw/go/crystal imports [github.com/ebitengine/purego]
FAIL
FAIL	go/build	2.835s
...
--- FAIL: TestDependencyVersionsConsistent (0.00s)
    moddeps_test.go:415: Modules within GOROOT require different versions of golang.org/x/sys.
    moddeps_test.go:427: std	requires v0.24.0
    moddeps_test.go:427: cmd	requires v0.15.0
FAIL
FAIL	cmd/internal/moddeps	4.626s
...
```

For the `TestDependencies`, I can solve the problem by add rules for importing purego.
But for the `TestDependencyVersionsConsistent`, Go complains that `golang.org/x/sys` must be consistent (to v0.15.0)

--- a/examples/libc/main_windows.go
+++ b/examples/libc/main_windows.go
@@ -3,9 +3,11 @@
 
 package main
 
-import "golang.org/x/sys/windows"
+import "syscall"
 
 func openLibrary(name string) (uintptr, error) {
-	handle, err := windows.LoadLibrary(name)
+	// Use [syscall.LoadLibrary] here to avoid external dependencies (#270).
+	// For actual use cases, [golang.org/x/sys/windows.NewLazySystemDLL] is recommended.
+	handle, err := syscall.LoadLibrary(name)
 	return uintptr(handle), err
 }

--- a/examples/window/main_windows.go
+++ b/examples/window/main_windows.go
@@ -5,9 +5,8 @@ package main
 
 import (
 	"runtime"
+	"syscall"
 	"unsafe"
-
-	"golang.org/x/sys/windows"
 
 	"github.com/ebitengine/purego"
 )
@@ -78,10 +77,14 @@ var (
 )
 
 func init() {
-	kernel32 := windows.NewLazySystemDLL("kernel32.dll").Handle()
+	// Use [syscall.NewLazyDLL] here to avoid external dependencies (#270).
+	// For actual use cases, [golang.org/x/sys/windows.NewLazySystemDLL] is recommended.
+	kernel32 := syscall.NewLazyDLL("kernel32.dll").Handle()
 	purego.RegisterLibFunc(&GetModuleHandle, kernel32, "GetModuleHandleW")
 
-	user32 := windows.NewLazySystemDLL("user32.dll").Handle()
+	// Use [syscall.NewLazyDLL] here to avoid external dependencies (#270).
+	// For actual use cases, [golang.org/x/sys/windows.NewLazySystemDLL] is recommended.
+	user32 := syscall.NewLazyDLL("user32.dll").Handle()
 	purego.RegisterLibFunc(&RegisterClassEx, user32, "RegisterClassExW")
 	purego.RegisterLibFunc(&CreateWindowEx, user32, "CreateWindowExW")
 	purego.RegisterLibFunc(&AdjustWindowRect, user32, "AdjustWindowRect")
@@ -96,7 +99,7 @@ func init() {
 }
 
 func main() {
-	className, err := windows.UTF16PtrFromString("Sample Window Class")
+	className, err := syscall.UTF16PtrFromString("Sample Window Class")
 	if err != nil {
 		panic(err)
 	}
@@ -104,7 +107,7 @@ func main() {
 
 	wc := WNDCLASSEX{
 		Size:      uint32(unsafe.Sizeof(WNDCLASSEX{})),
-		WndProc:   windows.NewCallback(wndProc),
+		WndProc:   syscall.NewCallback(wndProc),
 		Instance:  inst,
 		ClassName: className,
 	}
@@ -117,7 +120,7 @@ func main() {
 		Right:  320,
 		Bottom: 240,
 	}
-	title, err := windows.UTF16PtrFromString("My Title")
+	title, err := syscall.UTF16PtrFromString("My Title")
 	if err != nil {
 		panic(err)
 	}
@@ -130,7 +133,7 @@ func main() {
 		0, 0, inst, nil,
 	)
 	if hwnd == 0 {
-		panic(windows.GetLastError())
+		panic(syscall.GetLastError())
 	}
 
 	ShowWindow(hwnd, SW_SHOW)

--- a/go.mod
+++ b/go.mod
@@ -1,5 +1,3 @@
 module github.com/ebitengine/purego
 
 go 1.18
-
-require golang.org/x/sys v0.23.0

